### PR TITLE
Fix blank recipe editor on cancel save changes

### LINKED ISSUE #134: Clicking Cancel with unsaved changes in RecEditor blanks the RecEditor
Clicking Cancel with unsaved changes in the RecEditor will close the window, and will display a blank window when reopening it.

The cancel action should not close the window to begin with, but it seems that it increments the ingredient count, but is not saved in the `IngredientController.imodel`.

Below is a recoding showing the issue:
rec_editor_cancel

It's a rather easy fix: the issue can be mitigated by doing nothing on cancellation (ie. do nothing when handling `UserCancelledError`, [as is done in the timer](https://github.com/kirienko/gourmet/blob/40e911ab1369d4fcbf78ba1895326dd911f34b33/gourmet/timer.py#L246))

### PR DESCRIPTION
**Pull request by cydanil**
This closes #134 , where it's described how the Recipe Editor may get blank when not saving changes.

I'm not sure why `True` has to be returned (as opposed to `False` which is fine in the timer), but that fixes the behaviour as intended.

Thanks!

### CODE REVIEW

**Comment by maweki** [comment]
I think the true/false on signal handling is whether the signal is done being handled or whether it should continue being handled by other or the same callbacks.

File: gourmet/reccard.py [hunk 1]
```diff
@@ -1054,7 +1054,7 @@ def revert_cb (self, *args):
     def delete_cb (self, *args):
         self.rg.rec_tree_delete_recs([self.current_rec])
 
-    def close_cb (self, *args):
+    def close_cb(self, *args: Tuple[Gtk.Window, Gdk.Event]) -> bool:
```
Comment (maweki) [inline gourmet/reccard.py:1058]: If we already know that it's only two values, why `*args`?

Comment (cydanil) [reply gourmet/reccard.py:1058]: I didn't change what was already there, and as both values are ignored, I considered it good enough.

Comment (maweki) [reply gourmet/reccard.py:1058]: Okay. Keep it that way. I will change it, if `mypy` complains. :)

**Comment by cydanil** [comment]
If that's fine, I'll merge tomorrow :)

**Pull request merged**